generic/l10n_th_account_billing/models/account_voucher.py

```diff
@@ -32,34 +32,42 @@ class AccountVoucher(models.Model):
         readonly=True,
         states={'draft': [('readonly', False)]})
 
-    @api.one
+    @api.multi
     def proforma_voucher(self):
         # Write payment id back to Billing Document
+        self.ensure_one()
```

Comment:
Please don't use ensure_one(), always use `for rec in self:`

Reply:
in original he use api.one(). if you sure this way is better i will follow you.